pkg/resources/secret_with_oauth_client_credentials_acceptance_test.go

```diff
@@ -0,0 +1,196 @@
+					importchecks.TestCheckResourceAttrInstanceState(helpers.EncodeResourceIdentifier(id), "api_authentication", integrationId.Name()),
+					importchecks.TestCheckResourceAttrInstanceState(helpers.EncodeResourceIdentifier(id), "oauth_scopes.#", "2"),
+					importchecks.TestCheckResourceAttrInstanceState(helpers.EncodeResourceIdentifier(id), "comment", ""),
+				),
+			},
+			// destroy
+			{
+				Config:  config.FromModel(t, secretModel),
+				Destroy: true,
+			},
+		},
+	})
+}
+
+func TestAcc_SecretWithClientCredentials_EmptyScopesList(t *testing.T) {
+	id := acc.TestClient().Ids.RandomSchemaObjectIdentifier()
+	name := id.Name()
+
+	integrationId := acc.TestClient().Ids.RandomAccountObjectIdentifier()
+	_, apiIntegrationCleanup := acc.TestClient().SecurityIntegration.CreateApiAuthenticationClientCredentialsWithRequest(t,
+		sdk.NewCreateApiAuthenticationWithClientCredentialsFlowSecurityIntegrationRequest(integrationId, true, "foo", "foo").
+			WithOauthAllowedScopes([]sdk.AllowedScope{{Scope: "foo"}, {Scope: "bar"}}),
+	)
+	t.Cleanup(apiIntegrationCleanup)
+
+	secretModel := model.SecretWithClientCredentials("s", integrationId.Name(), id.DatabaseName(), id.SchemaName(), name, []string{})
+
+	resource.Test(t, resource.TestCase{
+		ProtoV6ProviderFactories: acc.TestAccProtoV6ProviderFactories,
+		PreCheck:                 func() { acc.TestAccPreCheck(t) },
+		TerraformVersionChecks: []tfversion.TerraformVersionCheck{
+			tfversion.RequireAbove(tfversion.Version1_5_0),
+		},
+		CheckDestroy: acc.CheckDestroy(t, resources.SecretWithClientCredentials),
+		Steps: []resource.TestStep{
+			// create secret without providing oauth_scopes value
+			{
+				Config: config.FromModel(t, secretModel),
+				Check: resource.ComposeTestCheckFunc(
+					assert.AssertThat(t,
+						resourceassert.SecretWithClientCredentialsResource(t, secretModel.ResourceReference()).
+							HasNameString(name).
+							HasDatabaseString(id.DatabaseName()).
+							HasSchemaString(id.SchemaName()).
+							HasApiAuthenticationString(integrationId.Name()).
+							HasCommentString(""),
+					),
+					resource.TestCheckResourceAttr(secretModel.ResourceReference(), "oauth_scopes.#", "0"),
+				),
+			},
+			// Set oauth_scopes
+			{
+				Config: config.FromModel(t, secretModel.
+					WithOauthScopes([]string{"foo"}),
+				),
+				ConfigPlanChecks: resource.ConfigPlanChecks{
+					PreApply: []plancheck.PlanCheck{
+						planchecks.ExpectChange(secretModel.ResourceReference(), "oauth_scopes", tfjson.ActionUpdate, sdk.String("[]"), sdk.String("[foo]")),
+					},
+				},
+				Check: assert.AssertThat(t,
+					resourceassert.SecretWithClientCredentialsResource(t, secretModel.ResourceReference()).
+						HasNameString(name).
+						HasDatabaseString(id.DatabaseName()).
+						HasSchemaString(id.SchemaName()).
+						HasApiAuthenticationString(integrationId.Name()),
+					assert.Check(resource.TestCheckResourceAttr(secretModel.ResourceReference(), "oauth_scopes.#", "1")),
```

Comment:
You can add a custom method in the resourceassert package - check `_ext` files. This way we can assert slices, because generator doesn't handle them properly. There, you can compare slices by length, and then check if they have the same elements.

Reply:
Will be done in another time, for now I will keep the asserts as they are